app/models/spree/calculator/variable_percent_on_line_item.rb

```diff
@@ -0,0 +1,20 @@
+class Spree::Calculator::VariablePercentOnLineItem < Spree::Calculator
+  preference :minimum_discount, :decimal, default: 0
+  preference :minimum_discount_on_price_greater_than, :decimal, default: 0
+  preference :maximum_discount_on_price_greater_than, :decimal, default: 0
+  preference :maximum_discount, :decimal, default: 0
+  def self.description
```

Comment:
@rishabhguptarishi always leave spaces between method definations for better understanding.